snuba/query/snql/parser.py

```diff
@@ -1190,16 +1220,18 @@ def _array_column_conditions(
         return
 
     if isinstance(from_clause, QueryEntity):
-        _unpack_array_conditions(query, from_clause.schema)
+        _unpack_array_conditions(cast(LogicalQuery, query), from_clause.schema)
     elif isinstance(from_clause, JoinClause):
         alias_map = from_clause.get_alias_node_map()
         for alias, node in alias_map.items():
-            assert isinstance(node.data_source, QueryEntity)  # mypy
-            _unpack_array_conditions(query, node.data_source.schema, alias)
+            if isinstance(node.data_source, QueryEntity):  # mypy
+                _unpack_array_conditions(
+                    cast(LogicalQuery, query), node.data_source.schema, alias
+                )
```

Comment:
Note to self: this looks kinda weird, make sure it's not something that got lost in the shuffle